### ISSUE TITLE
Add ability to @mention a user

### ISSUE THREAD

**Issue by joshsmith** [code-corps/deprecated-code-corps-rails-api#54]
A user should be able to @mention users in both a `Post` and a `Comment`.

**Comment by joshsmith** [comment]
Regex for picking out users and teams is `(?:\W|^)@((?:(?:(?:[^-\W]-?))*)(?:[^\/\W]\/?)?(?:(?:(?:[^-\W]-?))*)\w+)$`

**Comment by joshsmith** [comment]
### Should work

```
@code-corps
@code-corps-corps
@code_corps
@be
@code-corps/team
@code-corps/team-name
@code-corps-corps/team-name-name
@c
@code3123/213123sd
```

### Should NOT work

```
@code---corps
example@codecorps.org
@code-corps-
@-code-corps
@codecorps///team
@code-/corps
@code/-corps
@organization/team/not-allowed
```

**Comment by joshsmith** [comment]
The user mentions returned by the API should individually have `indices` specifying where they're located in the text, the `username` and user's `name` along with `user_id`.

This will be similar for team mentions, except they will have `team_name`, `organization_name`, `organization_id`, and `team_id`, along with the team's full slug for assembling the URL.

**Comment by begedin** [comment]
**Blocked by #53**

**Comment by joshsmith** [comment]
Closed by #73 and #100
